activerecord/test/cases/store_test.rb

```diff
@@ -191,4 +191,42 @@ class StoreTest < ActiveRecord::TestCase
     second_dump = YAML.dump(loaded)
     assert_equal @john, YAML.load(second_dump)
   end
+
+  test "setting type casted values on build" do
+    jamie = Admin::User.new(name: "Jamie", active: "true", salary: 3.1999, birthday: '2000-01-01')
+    assert_equal true, jamie.active
+    assert_equal 3, jamie.salary
+    assert_equal Date.new(2000,1,1), jamie.birthday
+  end
+
+  test "setting type casted values from db" do
+    jamie = Admin::User.new(name: "Jamie")
+    jamie.active = "t"
+    jamie.salary = 3.1999 
+    jamie.birthday = '01/01/2000'
+    jamie.save!
+
+    jamie = Admin::User.where(name: "Jamie").first
+   
+    assert_equal '2000-01-01', jamie.json_data['birthday']    
```

Comment:
I've given this some thought, and I think we should continue to go with the solution of type casting the store, and updating the accessors to use `Hash#merge` rather than `Hash#[]=`. If a user manually assigns a string to a key in the hash, then there's nothing we can do about it. If they assign the hash itself, the values get casted.